### ISSUE TITLE
There is a noticeable lag after signing up as a supporter

### ISSUE THREAD

**Issue by rgroves** [learnwithjason/learnwithjason.dev-v2#14]
After signing up as a supporter on the Stripe checkout page, you get redirected back to the dashboard and it still looks like the screenshot below, showing the buttons to choose your level of support:

![Screenshot 2024-11-12 130610](https://github.com/user-attachments/assets/90fb4175-c80b-47c2-a880-132b18dc1772)

I didn't time how long it took until a refresh of the page showed the indicator that I was a supporter, but it was long enough for me to worry about if the supporter sign up process even worked, that is until a refresh yielded the indicator and page change shown in the screenshot below:

![Screenshot 2024-11-12 130901](https://github.com/user-attachments/assets/910a13fb-fae4-4980-bdec-2d117137f134)

Maybe an optimistic update of some kind is needed or an indicator for "supporter in progress" than can be used to hide the supporter options and/or display a message that indicates the supporter level may take some time to confirm and appear.

**Comment by jlengstorf** [comment]
yeah, this is a great call. I think maybe we can make that section a server island and do a lookup on each load, since the overall number of requests should be relatively low
